cirq/contrib/qcircuit/qcircuit_diagram.py

```diff
@@ -51,28 +55,61 @@ def _render(diagram: circuits.TextDiagramDrawer) -> str:
             post1 = r'\qw' if key in qw else ''
             post2 = r'\qwx' if key in qwx else ''
             diagram2.write(2*x + 2, y, post1 + post2)
-        diagram2.write(2*w - 1, y, r'&\qw\\')
+        diagram2.write(2 * w - 1, y, '\\\\')
     grid = diagram2.render(horizontal_spacing=0, vertical_spacing=0)
 
     output = '\\Qcircuit @R=1em @C=0.75em {\n \\\\\n' + grid + '\n \\\\\n}'
 
     return output
 
 
-def circuit_to_latex_using_qcircuit(
+def circuit_to_qcircuit_diagram(
         circuit: circuits.Circuit,
-        qubit_order: ops.QubitOrderOrList = ops.QubitOrder.DEFAULT) -> str:
-    """Returns a QCircuit-based latex diagram of the given circuit.
+        *,
+        optimizers: Iterable[
+            Callable[[circuits.Circuit], None]] = default_optimizers,
+        get_circuit_diagram_info: Optional[
+            Callable[[ops.Operation, protocols.CircuitDiagramInfoArgs],
+                     protocols.CircuitDiagramInfo]] = None,
+        qubit_namer: Optional[Callable[[ops.Qid], str]] = None,
```

Comment:
The number of these arguments that are of the form "special case this specific object's behavior for this specific case" is indicative that we should perhaps be defining a protocol for latex output.